ethereum/jsonrpc/src/test/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/EthHashrateTest.java

```diff
@@ -0,0 +1,86 @@
+ */
+package tech.pegasys.pantheon.ethereum.jsonrpc.internal.methods;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+import tech.pegasys.pantheon.ethereum.blockcreation.EthHashMiningCoordinator;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.JsonRpcRequest;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcError;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcErrorResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcSuccessResponse;
+
+import java.util.Optional;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EthHashrateTest {
+
+  @Mock private EthHashMiningCoordinator miningCoordinator;
+  private EthHashrate method;
+  private final String JSON_RPC_VERSION = "2.0";
+  private final String ETH_METHOD = "eth_hashrate";
+
+  @Before
+  public void setUp() {
+    method = new EthHashrate(miningCoordinator);
+  }
+
+  @Test
+  public void returnsCorrectMethodName() {
+    assertThat(method.getName()).isEqualTo(ETH_METHOD);
+  }
+
+  @Test
+  public void shouldReturnValueFromMiningCoordinator() {
+    final JsonRpcRequest request = requestWithParams();
+    final JsonRpcResponse expectedResponse = new JsonRpcSuccessResponse(request.getId(), "0xc");
+    when(miningCoordinator.hashesPerSecond()).thenReturn(Optional.of(12L));
+
+    final JsonRpcResponse actualResponse = method.response(request);
+    assertThat(actualResponse).isEqualToComparingFieldByField(expectedResponse);
+  }
+
+  @Test
+  public void shouldReturnErrorWhenMiningCoordinatorDoesNotHaveHashes() {
+    final JsonRpcRequest request = requestWithParams();
+    final JsonRpcResponse expectedResponse =
+        new JsonRpcErrorResponse(request.getId(), JsonRpcError.NO_HASHES_PER_SECOND);
+    when(miningCoordinator.hashesPerSecond()).thenReturn(Optional.empty());
+
+    final JsonRpcResponse actualResponse = method.response(request);
+    assertThat(actualResponse).isEqualToComparingFieldByField(expectedResponse);
+  }
+
+  @Test
+  public void shouldReturnErrorWhenMiningCoordinatorDoesNotSupportHashing() {
+    final JsonRpcRequest request = requestWithParams();
+    final JsonRpcResponse expectedResponse =
+        new JsonRpcErrorResponse(request.getId(), JsonRpcError.INVALID_REQUEST);
+    when(miningCoordinator.hashesPerSecond()).thenThrow(UnsupportedOperationException.class);
+
+    final JsonRpcResponse actualResponse = method.response(request);
+    assertThat(actualResponse).isEqualToComparingFieldByField(expectedResponse);
+  }
+
+  private JsonRpcRequest requestWithParams() {
+    return new JsonRpcRequest(JSON_RPC_VERSION, ETH_METHOD, new Object[] {});
+  }
+}
```

Comment:
can I leave a comment on `ethereum/referencetests/src/test/resources`? not sure how to do that, anyway- probably don't want to change that in this pr